distributed/scheduler.py

```diff
@@ -3422,9 +3422,20 @@ def heartbeat_worker(
         ws._last_seen = time()
 
         if executing is not None:
-            ws._executing = {
-                parent._tasks[key]: duration for key, duration in executing.items()
-            }
+            ws._executing = {}
+            for key, duration in executing.items():
+                ts: TaskState = parent._tasks[key]
```

Comment:
@jakirkham Thanks for your pointer to the underscored attributes. I haven't been working with cython a lot so far and am wondering when I do need to declare types of my variables (probably always :P ) and when not. In particular I'm defining variables like old/new/avg_duration below. For full cythonization I would assume these should be declared as double or does cython pick this up automagically for primitive types?

Reply:
Ofc. Generally I try to type them. That said, Cython can pick up on things if one is assigning one typed variable to a new variable